### ISSUE TITLE
Faults are not lexed by mcompare

### ISSUE THREAD

**Issue by ambroise-arm** [herd/herdtools7#72]
Observed on version 7.56+02

With the following litmus test:
```
AArch64 Armv8-memtag
Orig=PodWRPA Amo.SwpAL RfeLP DMB.SYdRWPT CoeTP
{
0:X1=x:red; 0:X2=y:green;
1:X1=y:green; 1:X2=x:red; 1:X3=x:green;
}
 P0               | P1          ;
 MOV W0,#1        | LDR W0,[X1] ;
 STR W0,[X1]      | DMB SY      ;
 MOV W4,#1        | STG X2,[X3] ;
 SWPAL W4,W3,[X2] |             ;
exists (0:X3=0 /\ 1:X0=1) /\ ~fault(P0,x)
```

`herd7` outputs:
```
Test Armv8-memtag Allowed
States 3
0:X3=0; 1:X0=0;
0:X3=0; 1:X0=0; Fault(P0,x:red);
0:X3=0; 1:X0=1; Fault(P0,x:red);
No
Witnesses
Positive: 0 Negative: 3
Condition exists (0:X3=0 /\ 1:X0=1 /\ not (fault(P0,x)))
Observation Armv8-memtag Never 0 3
Time Armv8-memtag 0.08
Hash=fb94969c809eecdc7b5a41d38c9ec202
```

The previous output is stored in `out.log`. Doing `mcompare7 out.log out.log` gives:
```
File "out.log", line 4, character 15: Lex error pline
```

NB: when manually removing both `Fault(P0,x:red);` from out.log, `mcompare7` executes correctly, not having any problem with `fault(P0,x)` in the existence condition.

**Comment by maranget** [comment]
Hi, the KVM branch features a mcompare7 that handle faults. Modification is not simple, so I am reluctant to cherry-pick those into master. As a fix you can use the KVM branch. Notice that as far as you do not use option `-variant kvm` both branches should behave the same,

**Comment by maranget** [comment]
Issue solved by merging KVM branch into master, cf 8fd05a709999194ae452d8b72909c3fe58be3ed3.
